### DISCUSSION TITLE
What is the "injectionPoint"

### DISCUSSION THREAD

**Discussion by ioannis-tsirovasilis** [serwist/serwist#29] [Q&A]
Hello!

In the documentation https://serwist.pages.dev/docs/sw/install-serwist you state

 // Change this attribute's name to your `injectionPoint`. 

What exactly is this injectionPoint?

Thank you in advance!

**Comment by DuCanhGH** [comment]
`injectionPoint` is one of InjectManifest's options. Its value is searched and replaced with the list of precache entries in your service worker's source. By default, this value is set to `"self.__SW_MANIFEST"` (or `undefined` if you use `@serwist/vite`), hence the comment.

For example, if you use `@serwist/next`, this is how to configure it:
```js
import withSerwistInit from "@serwist/next";
      
const withSerwist = withSerwistInit({
    swSrc: "app/sw.ts",
    swDest: "public/sw.js",
    injectionPoint: "__HI_MOM"
});
         
export default withSerwist({
    // Your Next.js config
});
```

```ts
// app/sw.ts
import type { PrecacheEntry } from "@serwist/precaching";
import { installSerwist } from "@serwist/sw";
import { defaultCache } from "@serwist/next/browser";

declare const __HI_MOM: (PrecacheEntry | string)[] | undefined;

installSerwist({
  precacheEntries: __HI_MOM,
  skipWaiting: true,
  clientsClaim: true,
  navigationPreload: true,
  disableDevLogs: true,
  runtimeCaching: defaultCache,
});
```

**Reply by ioannis-tsirovasilis** [reply]
@DuCanhGH Thank you very much! So I can ignore it as it has a default value. I'v tried turning a NextJS 14 into a PWA, but when a upload to Heroku I get the following error (while building)

Module not found: Error: Can't resolve '/tmp/build_1c0ce433/src/app/sw.ts' in '/tmp/build_1c0ce433'
Can't find self.__SW_MANIFEST in your SW source.

This doesn't happen when running "npm run build" locally.

Have you ever faced something similar?

**Reply by DuCanhGH** [reply]
@ioannis-tsirovasilis huh, that looks like the plugin couldn't even resolve your `swSrc`. Not sure why it is trying to find '/tmp/build_\*/src/app/sw.ts' IN '/tmp/build_\*' though.

**Reply by ioannis-tsirovasilis** [reply]
@DuCanhGH Thank you! I hope to find what happens eventually

**Reply by hifilorau** [reply]
@ioannis-tsirovasilis @DuCanhGH did either of you guys ever figure this out? I'm having the same issue w next 14.  Have a pretty standard set up with sw.js in the root of my app folder along with my manifest file. 

The one thing to note though is in my case my build was originally failing to parse my manifest file when it was named manifest.json (def valid json with valid fields) but when i renamed it manifest.webmanifest it built fine. It even deployed once, but now i'm back in the same boat as Ioannis was above. 

I'lll also note when i run npm run build, it builds successfully and puts the sw.js file in the public folder. But when pushing to vercel...


```
Failed to compile.
--
16:00:05.620 |  
16:00:05.621 | Module not found: Error: Can't resolve '/vercel/path0/app/sw.js' in '/vercel/path0'

```

If it's helpful my sw.js file is currently only:

```
import { defaultCache } from "@serwist/next/browser";
import { installSerwist } from "@serwist/sw";

installSerwist({
  precacheEntries: self.__SW_MANIFEST,
  skipWaiting: true,
  clientsClaim: true,
  navigationPreload: true,
  disableDevLogs: true,
  runtimeCaching: defaultCache
});
```

**Reply by DuCanhGH** [reply]
@hifilorau never heard anything back from him, never bothered investigating either. Would you mind providing a reproduction?

**Reply by ioannis-tsirovasilis** [reply]
@hifilorau the problem was that I had added sw.ts to git ignore. Do you by any chance do the same thing?
